### PR TITLE
Fix to_dataframe() for operational query

### PR DESCRIPTION
**Pull request by dasjyao**
Many people have reported issues with `to_dataframe()` on operational statements, aka non-select queries. I think it would be good to fix this directly in locopy. The updated logic follows what [`snowflake_connector`](https://github.com/snowflakedb/snowflake-connector-python/blob/27f674aad6404e54bdd7c1f6531b7c4ee4b33385/src/snowflake/connector/cursor.py#L737-L742) does. An integration test has been added as well.

### CODE REVIEW

File: locopy/snowflake.py [hunk 1]
```diff
@@ -468,7 +468,7 @@ def to_dataframe(self, size=None):
             Dataframe with lowercase column names.  Returns None if no fetched
             result.
         """
-        if size is None:
+        if size is None and self.cursor._query_result_format == 'arrow':
```
Comment (jborchma) [inline locopy/snowflake.py:471]: So this now excludes all cases where `size` was none that were not in `arrow`-format. Those were the ones that are the result of an operational statement?

**Review by jborchma — COMMENTED** [review]
Just a quick question

File: locopy/snowflake.py (hunk 1 above)
Comment (dasjyao) [reply locopy/snowflake.py:471]: Yes, if you look at the page I shared from snowflake connector, and I also tested with several queries. Operational statements would have a Json-format.

**Comment by dasjyao** [comment]
Didn't expect something being impacted from unit test. I will fix this first.

File: locopy/snowflake.py (hunk 1 above)
Comment (jborchma) [reply locopy/snowflake.py:471]: Ah, makes sense

**Comment by jborchma** [comment]
So 5.3.1 seems to be the latest version. I don't think we can use that as the minimum...

**Comment by dasjyao** [comment]
> So 5.3.1 seems to be the latest version. I don't think we can use that as the minimum...

ahh makes a lot sense. @jborchma Since 5.3.1 failed WhiteSource check, can we bump it to ==5.3?

**Comment by jborchma** [comment]
I don't think that will do anything as I know for a fact that 5.3.0 and 5.2 have the same issue.

**Comment by jborchma** [comment]
Like internally, we might have to overrule the whitesource check here.

**Review by jborchma — APPROVED** [review]
LGTM

**Review by fdosani — APPROVED** [review]
LGTM also!

**Pull request merged**